dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -68,7 +71,8 @@ public void beforeStart(@Nullable UnitOfWork unitOfWork) {
             existingSession = sessionFactory.getCurrentSession();
         }
 
-        if (existingSession != null) {
+        if (existingSession != null && existingSession.isOpen() && existingSession.getSessionFactory() != null) {
```

Comment:
This session reuse block seemed like the most likely culprit for the issues we had, because it does not check that the session is in a good state, and there is no direct path to cleanup. I actually don't know in what states it hits this codepath, but seems like it could get stuck in a loop with a bad session that is never cleaned up which falls in line with our issues.